### PR TITLE
increase proto import timeout; bump golang to 1.21.1

### PR DESCRIPTION
**Pull request by tjons**
In bumping golang to 1.21 in `gloo-mesh-enterprise` (https://github.com/solo-io/gloo-mesh-enterprise/pull/11169) we saw consistent timeout issues with codegen only in CI that looked like this:

```
{
    "duration": {
        "@code-generator/proto-compiler": "9.691022611s"
generate.go:49: 9 errors occurred:
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/gloo-mesh-enterprise/api/gloo.solo.io/admin/v2alpha1/waypoint_lifecycle_manager.proto]
    },
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/gloo-mesh-enterprise/api/gloo.solo.io/apimanagement/v2/portal.proto]
    "frequency": {
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/gloo-mesh-enterprise/api/gloo.solo.io/admin/v2/gateway_lifecycle_manager.proto]
        "@code-generator/collector-opened-files": 241
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/gloo-mesh-enterprise/api/gloo.solo.io/admin/v2/istio_lifecycle_manager.proto]
    }
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/gloo-mesh-enterprise/api/gloo.solo.io/admin/v2/ratelimit_server_config.proto]
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/gloo-mesh-enterprise/api/gloo.solo.io/policy/v2/security/ext_auth_policy.proto]
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/gloo-mesh-enterprise/api/gloo.solo.io/networking/v2/route_table.proto]
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/solo-apis/api/gloo/enterprise.gloo/v1/auth_config.proto]
	* Timed out while fetching imports for proto file: [/home/runner/work/gloo-mesh-enterprise/gloo-mesh-enterprise/vendor_any/github.com/solo-io/solo-kit/api/external/envoy/api/v2/discovery.proto]
```

To resolve, I traced the issue back to skv2 and realized that we had a hard-coded 5 second limit for resolving proto imports. The code in this repo indicated that this error was only possible with circular dependencies, but since I could successfully compile our protos locally I increased the timeout to 15s and it worked as expected. I'm not sure of the root cause here - likely a performance regression in go1.21 or some of the related toolchain, but we need to resolve this for now so we can complete the upgrade. I also bumped go to 1.21.1 in this repo so that the projects are using the same golang versions.

### CODE REVIEW

**Comment by EItanya** [comment]
>  I'm not sure of the root cause here - likely a performance regression in go1.21 or some of the related toolchain, but we need to resolve this for now so we can complete the upgrade

This feels like something we need to resolve before upgrading, or at least being 100% certain this doesn't impact the product as well. Can we run perf testing for the product?

**Comment by tjons** [comment]
@EItanya - yeah was planning on running a perf test before we merge the upgrade into gloo-platform. I can do that before we merge this PR too if you'd like.

**Comment by EItanya** [comment]
I think I'd prefer that if it's ok. That way we can continue to make changes to skv2 as needed in the meantime.

**Review by ashleywang1 — APPROVED** [review]

**Comment by tjons** [comment]
Cool, sounds good to me @EItanya. Converted to draft.

**Review by dmitri-d — APPROVED** [review]

**Comment by EItanya** [comment]
Oh wait, I realized I'm a dummy and that upgrading this will not affect anyone downstream because each project needs to update to go 1.21 itself. Will undraft

**Comment by tjons** [comment]
I ran performance tests on this last night: https://github.com/solo-io/gloo-mesh-enterprise/pull/11169#issuecomment-1737513278

**Pull request merged**
